pyqt-apps/siriushla/widgets/state_button.py

```diff
@@ -1429,11 +1429,15 @@ def __init__(self, parent=None, init_channel=None):
         self.shape = 0
         self.renderer = QSvgRenderer()
 
-    def mousePressEvent(self, ev):
+    def mouseReleaseEvent(self, ev):
         """Deal with mouse clicks. Only accept clicks within the figure."""
-        if (ev.x() < self.width()/2+self.height() and
-                ev.x() > self.width()/2-self.height()):
+        cond = ev.button() == Qt.LeftButton
+        cond &= ev.x() < self.width()/2+self.height()
+        cond &= ev.x() > self.width()/2-self.height()
+        if cond:
             self.clicked.emit()
+        else:
+            super().mouseReleaseEvent(ev)
```

Comment:
I have only one question: why is this event not always emitted?

Reply:
well pointed @anacso17 . Actually I think it is not necessary to propagate this event to to the base class at all. I will remove this.